haskell/copy-dep-haddock.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+#
+# Usage: copy-dep-haddock.sh
+#
+# Environment variables:
+#   * RULES_HASKELL_MKDIR -- location of mkdir
+#   * RULES_HASKELL_CP -- location of cp
+#   * RULES_HASKELL_DOC_DIR -- root of doc directory
+#   * RULES_HASKELL_HTML_DIR -- html directory with Haddocks to copy
+#   * RULES_HASKELL_TARGET_DIR -- directory where to copy contents of html dir
+
+set -o pipefail
+
+# Ensure that top-level doc directory exists.
+
+"$RULES_HASKELL_MKDIR" -p "$RULES_HASKELL_DOC_DIR"
```

Comment:
This is a pattern we use elsewhere too, as an alternative to building a `PATH` that could introduce impurities. However, I'd be curious to know, what's the idiomatic way to do this in Bazel best practices? Are there established best practices already (in particular by ex-Blaze users) on how to feed scripts with their dependencies? Would it be better to use runfiles instead? Question for @judah.

Reply:
Generally, if a shell script depends on another binary or file, they'll declare it as a runfile and then reference it within the script using a hard-coded relative path.  The path to the script's directory (and thus to the runfiles) is obtained using something like:
```$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"```
which is more robust to symlinks.

However, for binaries like "mkdir" or "cp", all the other scripts I've seen have just called them directly, assuming they would be on the default path.  For example:
https://github.com/bazelbuild/bazel/search?utf8=%E2%9C%93&q=mkdir&type=
I guess that only really becomes an issue when you're relying on Nix to supply those binaries.

Honestly, though, for something as simple as this script I'd feel it simpler to just call `ctx.actions.run_shell`.  Both approaches seem defensible though.

Reply:
> I guess that only really becomes an issue when you're relying on Nix to supply those binaries.

Right. On NixOS, @mrkkrp has no alternative but to do what he did or to use runfiles. `/bin/` just doesn't exist.